itests/sponsorship_tx_test.go

```diff
@@ -0,0 +1,132 @@
+				utl.WavesDiffBalanceCheck(suite.T(), td.Expected.WavesDiffBalance, actualDiffBalanceInWaves.BalanceInWavesGo,
+					actualDiffBalanceInWaves.BalanceInWavesScala, utl.GetTestcaseNameWithVersion(name, v))
+				utl.AssetDiffBalanceCheck(suite.T(), td.Expected.AssetDiffBalance, actualDiffBalanceInAsset.BalanceInAssetGo,
+					actualDiffBalanceInAsset.BalanceInAssetScala, utl.GetTestcaseNameWithVersion(name, v))
+			})
+		}
+	}
+}
+
+func (suite *SponsorshipTxSuite) TestSponsorshipOffTx() {
+	versions := sponsor_utilities.GetVersions()
+	name := "Sponsorship On/Off"
+	waitForTx := true
+	for _, v := range versions {
+		reissuable := testdata.GetCommonIssueData(&suite.BaseSuite).Reissuable
+		itx := issue_utilities.IssueSendWithTestData(&suite.BaseSuite, reissuable, v, waitForTx)
+		sponsorship := testdata.GetSponsorshipOnOffData(&suite.BaseSuite, itx.TxID)
+		suite.Run(utl.GetTestcaseNameWithVersion(name, v), func() {
+			//switch on sponsorship
+			tx, actualDiffBalanceInWaves, actualDiffBalanceInAsset := sponsor_utilities.SendSponsorshipTxAndGetBalances(
+				&suite.BaseSuite, sponsorship.On, v, waitForTx)
+
+			utl.TxInfoCheck(suite.T(), tx.WtErr.ErrWtGo, tx.WtErr.ErrWtScala, "Sponsorship: "+tx.TxID.String(),
+				utl.GetTestcaseNameWithVersion(name, v))
+			utl.WavesDiffBalanceCheck(suite.T(), sponsorship.On.Expected.WavesDiffBalance, actualDiffBalanceInWaves.BalanceInWavesGo,
+				actualDiffBalanceInWaves.BalanceInWavesScala, utl.GetTestcaseNameWithVersion(name, v))
+			utl.AssetDiffBalanceCheck(suite.T(), sponsorship.On.Expected.AssetDiffBalance, actualDiffBalanceInAsset.BalanceInAssetGo,
+				actualDiffBalanceInAsset.BalanceInAssetScala, utl.GetTestcaseNameWithVersion(name, v))
+
+			//switch off sponsorship
+			tx, actualDiffBalanceInWaves, actualDiffBalanceInAsset = sponsor_utilities.SendSponsorshipTxAndGetBalances(
+				&suite.BaseSuite, sponsorship.Off, v, waitForTx)
+
+			utl.TxInfoCheck(suite.T(), tx.WtErr.ErrWtGo, tx.WtErr.ErrWtScala, "Sponsorship Off: "+tx.TxID.String(),
+				utl.GetTestcaseNameWithVersion(name, v))
+			utl.WavesDiffBalanceCheck(suite.T(), sponsorship.Off.Expected.WavesDiffBalance, actualDiffBalanceInWaves.BalanceInWavesGo,
+				actualDiffBalanceInWaves.BalanceInWavesScala, utl.GetTestcaseNameWithVersion(name, v))
+			utl.AssetDiffBalanceCheck(suite.T(), sponsorship.Off.Expected.AssetDiffBalance, actualDiffBalanceInAsset.BalanceInAssetGo,
+				actualDiffBalanceInAsset.BalanceInAssetScala, utl.GetTestcaseNameWithVersion(name, v))
+		})
+	}
+}
+
+func (suite *SponsorshipTxSuite) TestSponsorshipTxNegative() {
+	versions := sponsor_utilities.GetVersions()
+	waitForTx := true
+	for _, v := range versions {
+		reissuable := testdata.GetCommonIssueData(&suite.BaseSuite).Reissuable
+		itx := issue_utilities.IssueSendWithTestData(&suite.BaseSuite, reissuable, v, waitForTx)
+		tdmatrix := testdata.GetSponsorshipNegativeDataMatrix(&suite.BaseSuite, itx.TxID)
+		txIds := make(map[string]*crypto.Digest)
```

Comment:
I can't fined any usages of this variable except this one. Is `txIds` really necessary?

Reply:
Same as above, fix it.